servers/tests/org.fusesource.ide.server.karaf.core.tests/src-test/org/fusesource/ide/server/karaf/core/server/subsystems/KarafUtilsTest.java

```diff
@@ -0,0 +1,126 @@
+		version = KarafUtils.getBundleVersionFromFileName(bundleFile, KarafUtils.PACKAGING_BUNDLE);
+		assertEquals("1.0.0", version);
+		
+		bundleFile = new File("./target/test-1.0.0-SNAPSHOT.jar");
+		version = KarafUtils.getBundleVersionFromFileName(bundleFile, KarafUtils.PACKAGING_BUNDLE);
+		assertEquals("1.0.0.SNAPSHOT", version);
+		
+		bundleFile = new File("./target/test-1.0.0.war");
+		version = KarafUtils.getBundleVersionFromFileName(bundleFile, KarafUtils.PACKAGING_WAR);
+		assertEquals("1.0.0", version);
+		
+		bundleFile = new File("./target/test-1.0.0-SNAPSHOT.war");
+		version = KarafUtils.getBundleVersionFromFileName(bundleFile, KarafUtils.PACKAGING_WAR);
+		assertEquals("1.0.0.SNAPSHOT", version);
+	}
+	
+	@Test
+	public void testGetBundleVersionFromURI() throws CoreException {
+		File bundleFile = new File("./target/test-1.0.0.jar");
+		String uri = String.format("%sfile:%s", getProtocolPrefixForModule(KarafUtils.PACKAGING_BUNDLE), bundleFile.toURI().toString());
+		String version = KarafUtils.getBundleVersionFromURI(uri, KarafUtils.PACKAGING_BUNDLE);
+		assertEquals("1.0.0", version);
+		
+		bundleFile = new File("./target/test-1.0.0-SNAPSHOT.jar");
+		uri = String.format("%sfile:%s", getProtocolPrefixForModule(KarafUtils.PACKAGING_BUNDLE), bundleFile.toURI().toString());
+		version = KarafUtils.getBundleVersionFromURI(uri, KarafUtils.PACKAGING_BUNDLE);
+		assertEquals("1.0.0.SNAPSHOT", version);
+		
+		bundleFile = new File("./target/test.jar");
+		uri = String.format("%sfile:%s$Bundle-SymbolicName=%s&Bundle-Version=%s", getProtocolPrefixForModule(KarafUtils.PACKAGING_JAR), bundleFile.toURI().toString(), "test", "1.0.0");
+		version = KarafUtils.getBundleVersionFromURI(uri, KarafUtils.PACKAGING_JAR);
+		assertEquals("1.0.0", version);
+		
+		bundleFile = new File("./target/test.jar");
+		uri = String.format("%sfile:%s$Bundle-SymbolicName=%s&Bundle-Version=%s", getProtocolPrefixForModule(KarafUtils.PACKAGING_JAR), bundleFile.toURI().toString(), "test", "1.0.0.SNAPSHOT");
+		version = KarafUtils.getBundleVersionFromURI(uri, KarafUtils.PACKAGING_JAR);
+		assertEquals("1.0.0.SNAPSHOT", version);
+		
+		bundleFile = new File("./target/test.war");
+		uri = String.format("%sfile:%s?Bundle-SymbolicName=%s&Bundle-Version=%s", getProtocolPrefixForModule(KarafUtils.PACKAGING_WAR), bundleFile.toURI().toString(), "test", "1.0.0");
+		version = KarafUtils.getBundleVersionFromURI(uri, KarafUtils.PACKAGING_WAR);
+		assertEquals("1.0.0", version);
+		
+		bundleFile = new File("./target/test.war");
+		uri = String.format("%sfile:%s?Bundle-SymbolicName=%s&Bundle-Version=%s", getProtocolPrefixForModule(KarafUtils.PACKAGING_WAR), bundleFile.toURI().toString(), "test", "1.0.0.SNAPSHOT");
+		version = KarafUtils.getBundleVersionFromURI(uri, KarafUtils.PACKAGING_WAR);
+		assertEquals("1.0.0.SNAPSHOT", version);
+	}
+	
+	
+	private File createTempManifest(String version) throws IOException {
+		File manifestFile = new File("./target/MANIFEST.MF");
```

Comment:
usually better to use TemporaryFolder JUnit rule when creating files, it avoids possible impacts on other tests (in case one is failing)